pkg/crc/pullsecret/pullsecret.go

```diff
@@ -0,0 +1,98 @@
+package pullsecret
+
+import (
+	"encoding/base64"
+	"fmt"
+
+	"github.com/code-ready/crc/pkg/crc/logging"
+	"github.com/code-ready/machine/libmachine/drivers"
+	"github.com/pborman/uuid"
+)
+
+var (
+	pullSecret = `apiVersion: v1
+data:
+  .dockerconfigjson: %s
+kind: Secret
+metadata:
+  name: pull-secret
+  namespace: openshift-config
+type: kubernetes.io/dockerconfigjson`
+
+	// This command timeout in 80 secs if not able to replace the pull
+	// secret. We first start the kubelet and wait till the api server is
+	// up and running by hitting it with replace command.
+	replacePullSecretCmd = `timeout 80 bash -c 'sudo systemctl start kubelet && \
+until oc --config /tmp/kubeconfig replace -f /tmp/pull-secret.yaml 2>/dev/null 1>&2; \
+do echo "Waiting for recovery apiserver to come up."; sleep 1; done'`
+
+	// This command is used to update the clusterID using random uuid.
+	updateClusterIdCmd = `oc --config /tmp/kubeconfig patch clusterversion version -p '{"spec":{"clusterID":"%s"}}' --type merge`
+
+	// This command make sure we stop the kubelet and clean up the pods
+	// in case there is an error or not.
+	stopKubeletAndRemovePodsCmd = `bash -c 'sudo systemctl stop kubelet && \
+sudo crictl stopp $(sudo crictl pods -q) && \
+sudo crictl rmp $(sudo crictl pods -q)'`
+)
+
+func AddPullSecretAndClusterID(driver drivers.Driver, pullSec string, kubeconfig string) (rerr error) {
+	defer func() {
+		output, err := drivers.RunSSHCommandFromDriver(driver, stopKubeletAndRemovePodsCmd)
+		logging.DebugF("Output of %s: %s", stopKubeletAndRemovePodsCmd, output)
+		if err != nil {
+			rerr = err
+		}
+	}()
+
+	if err := addKubeconfigFileToInstance(driver, kubeconfig); err != nil {
+		return err
+	}
+	if err := addpullSecretFileToInstance(driver, pullSec); err != nil {
+		return err
+	}
+	output, err := drivers.RunSSHCommandFromDriver(driver, replacePullSecretCmd)
+	if err != nil {
+		return err
+	}
+	logging.DebugF("Output of %s: %s", replacePullSecretCmd, output)
+
+	clusterID := uuid.New()
+	output, err = drivers.RunSSHCommandFromDriver(driver, fmt.Sprintf(updateClusterIdCmd, clusterID))
+	if err != nil {
+		return err
+	}
```

Comment:
This can probably be rearranged in:
- start kubelet
- call addPullSecret method which will copy the file and run the needed command
- call addClusterId method which will copy the file and ru the needed command
- stop kubelet (maybe in defer)

Reply:
@cfergeau doesn't this we are doing as of now, just kubelet start and add pull secret is in same command.

Reply:
yes, kubelet start would be better in a separate command, and the subfunctions/methods did not exist in the version I reviewed.

Reply:
@cfergeau yes I updated those now.